src/index.js

```diff
@@ -0,0 +1,898 @@
+      this.searchItemPressed(selectedItem);
+
+      return;
+    }
+
+    /**
+     * Get input field value
+     */
+    const href = this.nodes.inputField.value;
+
+    /**
+     * If input field is empty then do nothing
+     */
+    if (!href || !href.trim()) {
+      return;
+    }
+
+    /**
+     * If input is not a valid url then show an error
+     */
+    if (!url(href)) {
+      notifier.show({
+        message: DICTIONARY.invalidUrl,
+        style: 'error',
+      });
+
+      return;
+    }
+
+    /**
+     * Get the first item from the search list
+     * This item exists because input href is valid
+     *
+     * @type {Element}
+     */
+    const composedItem = this.getSearchItems()[0];
+
+    /**
+     * "Press" search item
+     */
+    this.searchItemPressed(composedItem);
+  }
+
+  /**
+   * Get search items
+   *
+   * @returns {Element[]}
+   */
+  getSearchItems() {
+    const nodesList = this.nodes.searchResults.querySelectorAll(`.${this.CSS.searchItem}`);
+
+    return Array.from(nodesList);
+  }
+
+  /**
+   * Find selected item
+   *
+   * @returns {Element|null}
+   */
+  getSelectedItem() {
+    return this.nodes.searchResults.querySelector(`.${this.CSS.searchItemSelected}`);
+  }
+
+  /**
+   * Remove search result elements
+   */
+  clearSearchList() {
+    while (this.nodes.searchResults.firstChild) {
+      this.nodes.searchResults.firstChild.remove();
+    }
+  }
+
+  /**
+   * Fill up a search list results by data
+   *
+   * @param {SearchItemData[]} items
+   */
+  generateSearchList(items = []) {
+    /**
+     * Clear list first
+     */
+    this.clearSearchList();
+
+    /**
+     * If items data is not an array
+     */
+    if (!Utils.isArray(items)) {
+      notifier.show({
+        message: DICTIONARY.invalidServerData,
+        style: 'error',
+      });
+
+      return;
+    }
+
+    /**
+     * If no items returned
+     */
+    if (items.length === 0) {
+      return;
+    }
+
+    /**
+     * Fill up search list by new elements
+     */
+    items.forEach(item => {
+      const searchItem = Dom.make('DIV', [ this.CSS.searchItem ]);
+
+      /**
+       * Create a name for a link
+       *
+       * @type {HTMLElement}
+       */
+      const searchItemName = Dom.make('DIV', [ this.CSS.searchItemName ], {
+        innerText: item.name || item.href,
+      });
+
+      searchItem.appendChild(searchItemName);
+
+      /**
+       * Create a description element
+       */
+      if (item.description) {
+        const searchItemDescription = Dom.make('DIV', [ this.CSS.searchItemDescription ], {
+          innerText: item.description,
+        });
+
+        searchItem.appendChild(searchItemDescription);
+      }
+
+      /**
+       * Save all keys to item's dataset
+       */
+      Object.keys(item).forEach(key => {
+        searchItem.dataset[key] = item[key];
+      });
+
+      /**
+       * Enable click listener
+       */
+      searchItem.addEventListener('click', (event) => {
```

Comment:
you can optimize memory by providing the single click listener on the search list wrapper and detecting clicked item by `event.target`

[Example](https://github.com/editor-js/table/pull/72/files#diff-0921e2927cac9e25d54b12dd523d94bc87717425a4d1202bf1b7e7e7ede9a0beR85-R105)